src/pages/ReportParticipantsPage.tsx

```diff
@@ -1,75 +1,82 @@
 
-const propTypes = {
-    /* Onyx Props */
-
-    /** The personal details of the person who is logged in */
-    personalDetails: PropTypes.objectOf(personalDetailsPropType),
-
-    /** The active report */
-    report: reportPropTypes.isRequired,
-
-    /** Route params */
-    route: PropTypes.shape({
-        params: PropTypes.shape({
-            /** Report ID passed via route r/:reportID/participants */
-            reportID: PropTypes.string,
-        }),
-    }).isRequired,
-
-    ...withLocalizePropTypes,
+type ReportParticipantsPageOnyxProps = {
+    personalDetails: OnyxCollection<PersonalDetails>;
 };
 
-const defaultProps = {
-    personalDetails: {},
-};
+type ReportParticipantsPageProps = ReportParticipantsPageOnyxProps & WithReportOrNotFoundProps;
+
+type ParticipantData = {
+    alternateText: string;
+    displayName: string;
+    accountID: number;
+    icons: Array<{
+        id: number;
+        source: UserUtils.AvatarSource;
+        name: string;
+        type: ValueOf<typeof CONST.ICON_TYPE_AVATAR>;
+    }>;
+    keyForList: string;
+    login: string;
+    text: string;
+    tooltipText: string;
+    participantsList: Array<{
+        accountID: number;
+        displayName: string;
+    }>;            
+}
 
 /**
  * Returns all the participants in the active report
  *
- * @param {Object} report The active report object
- * @param {Object} personalDetails The personal details of the users
- * @param {Object} translate The localize
- * @return {Array}
+ * @param report The active report object
+ * @param personalDetails The personal details of the users
+ * @param translate The localize
  */
-const getAllParticipants = (report, personalDetails, translate) =>
-    _.chain(ReportUtils.getVisibleMemberIDs(report))
+const getAllParticipants = (
+    report: OnyxEntry<Report>,
+    personalDetails: OnyxCollection<PersonalDetails>,
+    translate: <TKey extends TranslationPaths>(phraseKey: TKey, ...phraseParameters: Localize.PhraseParameters<Localize.Phrase<TKey>>) => string,
+): ParticipantData[] =>
+    ReportUtils.getVisibleMemberIDs(report)
         .map((accountID, index) => {
-            const userPersonalDetail = lodashGet(personalDetails, accountID, {displayName: personalDetails.displayName || translate('common.hidden'), avatar: ''});
-            const userLogin = LocalePhoneNumber.formatPhoneNumber(userPersonalDetail.login || '') || translate('common.hidden');
+            const userPersonalDetail = personalDetails?.[accountID];
+            const userLogin = LocalePhoneNumber.formatPhoneNumber(userPersonalDetail?.login ?? '') ?? translate('common.hidden');
             const displayName = PersonalDetailsUtils.getDisplayNameOrDefault(userPersonalDetail);
 
             return {
                 alternateText: userLogin,
                 displayName,
-                accountID: userPersonalDetail.accountID,
+                accountID: userPersonalDetail?.accountID ?? accountID,
```

Comment:
is it correct to use `accountID` here?

Reply:
I think so, I tested it and it all works properly. Previously there was no fallback (js didn't mind)